### PR TITLE
Add new NPS survey

### LINKED ISSUE #1854: Start Survey to get user feedback and satisfaction
This survey should not start until June 1st.

### PR DESCRIPTION
**Pull request by bwateratmsft**
Fixes #1854 (will need cherry-pick to master)

### CODE REVIEW

File: src/telemetry/surveys/nps2.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ *  Licensed under the MIT License. See License.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { localize } from '../../localize';
+import { Survey } from './SurveyManager';
+
+const nps2: Survey = {
+    id: 'nps2',
+    url: 'https://microsoft.com', // TODO
```
Comment (bwateratmsft) [inline src/telemetry/surveys/nps2.ts:12]: Need the new URL

File: src/telemetry/surveys/nps2.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    prompt: localize('vscode-docker.survey.nps.prompt', 'Do you mind taking a quick feedback survey about the Docker Extension for VS Code?'),
```
Comment (bwateratmsft) [inline src/telemetry/surveys/nps2.ts:13]: @BigMorty for final text

File: src/telemetry/surveys/SurveyManager.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+            }
+        } catch { } // Best effort
+    }
+
+    private async surveyOpen(survey: Survey): Promise<void> {
+        await vscode.env.openExternal(vscode.Uri.parse(`${survey.url}?o=${encodeURIComponent(process.platform)}&m=${encodeURIComponent(vscode.env.machineId)}`));
+    }
+
+    private async surveyPrompt(survey: Survey): Promise<boolean> {
+        await ext.context.globalState.update(`${surveyRespondedKeyPrefix}.${survey.id}`, true);
```
Comment (bwateratmsft) [inline src/telemetry/surveys/SurveyManager.ts:65]: This is written but never used

File: src/telemetry/surveys/SurveyManager.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+        const take = localize('vscode-docker.survey.nps.take', 'Take Survey');
+        const never = localize('vscode-docker.survey.nps.never', 'Don\'t Ask Again');
+        const result = await vscode.window.showInformationMessage(survey.prompt, take, never);
+
+        return result === take;
+    }
+
+    private async shouldShowPrompt(survey: Survey): Promise<boolean> {
+        return ext.context.globalState.get<boolean>(`${surveyRespondedKeyPrefix}.${survey.id}`, false) !== true &&
```
Comment (karolz-ms) [inline src/telemetry/surveys/SurveyManager.ts:75]: Just curious: why `!== true` if the type is boolean?

File: src/telemetry/surveys/SurveyManager.ts [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+const currentSurveys = [nps2];
+
+const surveyRespondedKeyPrefix = 'vscode-docker.surveys.response';
+const surveyFlightPrefix = 'vscode-docker.surveys';
+
+export interface Survey {
+    id: string;
+    url: string;
+    prompt: string;
+    activationDelay: number;
```
Comment (karolz-ms) [inline src/telemetry/surveys/SurveyManager.ts:22]: Consider `activationDelayMs` for extra clarity (but feel free to disregard, I know not everyone likes this)

File: src/telemetry/surveys/SurveyManager.ts (hunk 4 above)
Comment (bwateratmsft) [reply src/telemetry/surveys/SurveyManager.ts:75]: Could also do `!ext...`, but it's meant to be `true` once they have responded, and we only show if they have never responded.

Comment (bwateratmsft) [reply src/telemetry/surveys/SurveyManager.ts:75]: (For additional context, `undefined`, which is default, is falsy, so I wanted both `undefined` and `false` to mean "hasn't shown the prompt yet")

**Pull request merged**